magma/protocol_type.py

```diff
@@ -67,5 +70,10 @@ def flatten(self):
     def name(self):
         return self._get_magma_value_().name
 
+    def wire(self, other):
+        if isinstance(other, MagmaProtocol):
+            other = other._get_magma_value_()
+        self._get_magma_value_().wire(other)
+
     def __imatmul__(self, other):
-        m.wire(other, self._get_magma_value_())
+        self.wire(other)
```

Comment:
Since it's `wire(o, i)` and in the case of imatmul, it's `i @= o`, I think this should be `other.wire(self)`

Reply:
Scratch that, that's the signature of wire, but then internally wire calls `o.wire(i)`